cpu/stm32/periph/pwm.c

```diff
@@ -29,12 +29,14 @@
 #include "periph/pwm.h"
 #include "periph/gpio.h"
 
-#define CCMR_LEFT           (TIM_CCMR1_OC1M_1 | TIM_CCMR1_OC1M_2 | \
+#define CCMR_MODE1          (TIM_CCMR1_OC1M_1 | TIM_CCMR1_OC1M_2 | \
                              TIM_CCMR1_OC2M_1 | TIM_CCMR1_OC2M_2)
-#define CCMR_RIGHT          (TIM_CCMR1_OC1M_0 | TIM_CCMR1_OC1M_1 | \
+#define CCMR_MODE2          (TIM_CCMR1_OC1M_0 | TIM_CCMR1_OC1M_1 | \
                              TIM_CCMR1_OC1M_2 | TIM_CCMR1_OC2M_0 | \
                              TIM_CCMR1_OC2M_1 | TIM_CCMR1_OC2M_2);
 
+static uint32_t dc_reverse = 0;
```

Comment:
This seems unused.

Reply:
This is used to store PWM devices which are initialized in PWM_RIGHT mode in the pwm_init function and whose duty cycle must be reversed by the pwm_set function.

Reply:
Now the pwm_set function looks into the CCMR hardware register instead.